### PR TITLE
`read_provider_test` migrate to devnet

### LINKED ISSUE #347: Migrate read_provider_test

### PR DESCRIPTION
**Pull request by tonypony220**
- updated `read_provider_test` to match 0.0.7 devnet
- fixed devnet starting command
- cleaned up melos commands
- adding devnet error to facilitate debugging
- some tests are waiting new release (0.0.8) of devnet and marked as FIXME in code

https://github.com/0xSpaceShard/starknet-devnet-rs/issues/498
https://github.com/0xSpaceShard/starknet-devnet-rs/issues/496
https://github.com/0xSpaceShard/starknet-devnet-rs/issues/544

resolves #347  and maybe #371

estimated time: 4 days 

## Summary by CodeRabbit

- **New Features**
  - Introduced `sum` method in smart contract for calculating the sum of two values.
  - Added new commands for `melos` to support smart contract transactions.

- **Enhancements**
  - Updated environment variables with new contract addresses for better deployment.
  - Improved error handling with new error codes in JSON-RPC.

- **Bug Fixes**
  - Modified test data values for accuracy in integration tests.

- **Documentation**
  - Added JSON configuration for deployment details.

- **Style**
  - Streamlined enum declaration for `TokenSymbol`.

- **Chores**
  - Updated `starknet-devnet` to version `0.0.7` across various files.
  - Added `.starkli/account_1.json` to `.gitignore`.

### CODE REVIEW

File: melos.yaml [hunk 1]
```diff
@@ -82,10 +83,16 @@ scripts:
...
-    description: Call hello get_name
+
+  contracts:invoke:
+    description: creating invoke tx for tests with hash 0x03b2911796e0024f9e23d7337997538058eca267d5ddaa582d482cbe1fb64897 and deploy_account tx with hash 0x055ba13c33a12506d2eab8dfbc618a8ce0d247c24959a64ee18fbf393c873b83
     run: |
-      starkli call $HELLO_CONTRACT_ADDRESS get_name
+      starkli invoke $HELLO_CONTRACT_ADDRESS set_name 0x0 
+      # should create tx hash 0x029583643cd8932f1955bf28bfebf4c907b13df1e5c2d202b133cfbf783697a2
+      
+      yes | cp .starkli/account_for_devnet_setup.json .starkli/account_1.json
```
Comment (ptisserand) [inline melos.yaml]: I suggest to use `-f` option of `cp` instead of spawning a `yes` process.

File: contracts/v2.6.2/src/hello.cairo [hunk 2]
```diff
@@ -25,5 +26,9 @@ mod hello {
         fn set_name(ref self: ContractState, name: felt252) {
             self.name.write(name);
         }
+
+        fn sum(ref self: ContractState, a: felt252, b: felt252) -> felt252 {
```
Comment (ptisserand) [inline contracts/v2.6.2/src/hello.cairo]: There is no modification of state, so you can use a snapshot instead:
```cairo
fn sum(self: @ContractState, a: felt252, b: felt252) -> felt252 {
```

File: melos.yaml [hunk 3]
```diff
@@ -82,10 +83,16 @@ scripts:
...
+      starkli invoke 0x49D36570D4E46F48E99674BD3FCC84644DDD6B96F7C741B1562B82F9E004DC7 transfer 0x016a0d7df981d681537dc2ce648722ff1d1c2cbe59412b492d35bac69825f104 0x100000000000000000 0x0
```
Comment (ptisserand) [inline melos.yaml]: It should be easier to read with `eth` instead of `0x49D...`

**Review by ptisserand — CHANGES_REQUESTED** [review]

File: contracts/v2.6.2/src/hello.cairo (hunk 2 above)
Comment (tonypony220) [reply contracts/v2.6.2/src/hello.cairo]: good point! thx

File: melos.yaml (hunk 3 above)
Comment (tonypony220) [reply melos.yaml]: done

File: contracts/v2.6.2/src/hello.cairo [hunk 4]
```diff
@@ -2,6 +2,7 @@
 trait IHello<T> {
     fn get_name(self: @T) -> felt252;
     fn set_name(ref self: T, name: felt252);
+    fn sum(ref self: T, a: felt252, b: felt252) -> felt252;
```
Comment (ptisserand) [inline contracts/v2.6.2/src/hello.cairo]: It should be a snapshot here instead of a ref.

**Review by ptisserand — CHANGES_REQUESTED** [review]

File: contracts/v2.6.2/src/hello.cairo (hunk 4 above)
Comment (tonypony220) [reply contracts/v2.6.2/src/hello.cairo]: ops! sorry

File: contracts/v2.6.2/src/hello.cairo [hunk 5]
```diff
@@ -2,7 +2,7 @@
 trait IHello<T> {
     fn get_name(self: @T) -> felt252;
     fn set_name(ref self: T, name: felt252);
-    fn sum(ref self: T, a: felt252, b: felt252) -> felt252;
+    fn sum(self: @ContractState, a: felt252, b: felt252) -> felt252;
```
Comment (ptisserand) [inline contracts/v2.6.2/src/hello.cairo]: Wrong copy&paste  ;) 
It should be:
```cairo
fn sum(self: @T, a: felt252, b: felt252) -> felt252;
```

**Review by ptisserand — CHANGES_REQUESTED** [review]
One last fix and we can merge!
Thanks a lot for your work

File: contracts/v2.6.2/src/hello.cairo (hunk 5 above)
Comment (tonypony220) [reply contracts/v2.6.2/src/hello.cairo]: oh no

**Review by ptisserand — APPROVED** [review]

**Pull request merged**